dangerzone/gvisor_wrapper/entrypoint.py

```diff
@@ -0,0 +1,267 @@
+        # over it.
+        {
+            "destination": "/home/dangerzone",
+            "type": "tmpfs",
+            "source": "tmpfs",
+            "options": ["nosuid", "noexec", "nodev"],
+        },
+        # Used for LibreOffice extensions, which are only conditionally
+        # installed depending on which file is being converted.
+        {
+            "destination": "/usr/lib/libreoffice/share/extensions/",
+            "type": "tmpfs",
+            "source": "tmpfs",
+            "options": ["nosuid", "noexec", "nodev"],
+        },
+    ],
+    "linux": {
+        "namespaces": [
+            {"type": "pid"},
+            {"type": "network"},
+            {"type": "ipc"},
+            {"type": "uts"},
+            {"type": "mount"},
+        ],
+    },
+}
+not_forwarded_env = set(("PATH", "HOME", "SHLVL", "HOSTNAME", "TERM", "PWD"))
+for key_val in oci_config["process"]["env"]:
+    not_forwarded_env.add(key_val[: key_val.index("=")])
+for key, val in os.environ.items():
+    if key in not_forwarded_env:
+        continue
+    oci_config["process"]["env"].append("%s=%s" % (key, val))
+if args.gvisor_debug:
+    print("Command inside gVisor sandbox:", wrapped_command, file=sys.stderr)
+    print("OCI config:", file=sys.stderr)
+    json.dump(oci_config, sys.stderr, indent=2, sort_keys=True)
+    # json.dump doesn't print a trailing newline, so print one here:
+    print("", file=sys.stderr)
+with open("/dangerzone-image/config.json", "w") as oci_config_out:
+    json.dump(oci_config, oci_config_out, indent=2, sort_keys=True)
+
+# Run gVisor.
+runsc_binary = "/usr/bin/runsc"
+runsc_argv = [os.path.basename(runsc_binary), "--rootless=true", "--network=none"]
+if args.gvisor_debug:
+    runsc_argv += ["--debug=true", "--alsologtostderr=true"]
+if args.gvisor_strace:
+    runsc_argv += ["--strace=true"]
+if args.gvisor_flag is not None:
+    for gvisor_flag in args.gvisor_flag:
+        if gvisor_flag:
+            runsc_argv += [gvisor_flag]
+runsc_argv += ["run", "--bundle=/dangerzone-image", "dangerzone"]
+
+# Check for `--pre_gvisor` which can be used to run commands without wrapping.
+if args.pre_gvisor:
+    print(
+        "Would be running the following command if --pre_gvisor had not been specified:",
+        " ".join(shlex.quote(s) for s in runsc_argv),
+        file=sys.stderr,
+    )
+    print(
+        "Executing this command instead:",
+        " ".join(shlex.quote(s) for s in wrapped_command),
+        file=sys.stderr,
+    )
+    try:
+        os.execvp(wrapped_command[0], wrapped_command)
+    except Exception as e:
+        raise e.__class__("Process %s failed: %s" % (wrapped_command, e))
+    else:
+        assert False, "This code should never be reachable"
+
+if args.gvisor_debug:
+    print(
+        "Running gVisor with command line:",
+        " ".join(shlex.quote(s) for s in runsc_argv),
+        file=sys.stderr,
+    )
+runsc_process = subprocess.run(
+    runsc_argv,
+    executable=runsc_binary,
```

Comment:
Minor, but what's the case for `executable=` here (and `os.path.basename(runsc_binary)` above)? Is it for stylistic reasons?